### PR TITLE
Refactor BuildIOSFrameworkCommand with common darwin baseclass

### PR DESCRIPTION
**Pull request by jmagman**
I am working on https://github.com/flutter/flutter/issues/104866 and want a common baseclass to share between `build ios-framework` and `build macos-framework`.  To keep that PR smaller, factor out the common properties and the xcframework logic that will be used by both.

Renamed test `build_ios_framework_test` to `build_darwin_framework_test` since the new macOS tests will live in that file, and if I change it now git will track the rename.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/bin/podhelper.rb [hunk 1]
```diff
@@ -177,9 +177,9 @@ def flutter_install_ios_engine_pod(ios_application_path = nil)
       Pod::Spec.new do |s|
         s.name             = 'Flutter'
         s.version          = '1.0.0'
-        s.summary          = 'High-performance, high-fidelity mobile apps.'
-        s.homepage         = 'https://flutter.io'
-        s.license          = { :type => 'MIT' }
+        s.summary          = 'A UI toolkit for beautiful and fast apps.'
+        s.homepage         = 'https://flutter.dev'
+        s.license          = { :type => 'BSD' }
```
Comment (jmagman) [inline packages/flutter_tools/bin/podhelper.rb:182]: This should be `BSD` https://github.com/CocoaPods/Specs/blob/master/Specs/c/0/4/Flutter/3.0.0/Flutter.podspec.json#L8

File: packages/flutter_tools/lib/src/commands/build_ios_framework.dart [hunk 2]
```diff
@@ -67,16 +64,6 @@ class BuildIOSFrameworkCommand extends BuildSubCommand {
         help: 'Whether to produce a framework for the release build configuration. '
               'By default, all build configurations are built.'
       )
-      ..addFlag('universal',
-        help: '(deprecated) Produce universal frameworks that include all valid architectures.',
-        hide: !verboseHelp,
-      )
-      ..addFlag('xcframework',
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/commands/build_ios_framework.dart:74]: These are both deprecated, don't add to new command that will inherit from `BuildIOSFrameworkCommand`.

File: packages/flutter_tools/lib/src/commands/build_ios_framework.dart [hunk 3]
```diff
@@ -23,26 +25,21 @@ import '../runner/flutter_command.dart' show DevelopmentArtifact, FlutterCommand
...
-    Platform? platform
+    Platform? platform,
   }) : _injectedFlutterVersion = flutterVersion,
        _buildSystem = buildSystem,
        _injectedCache = cache,
        _injectedPlatform = platform,
        super(verboseHelp: verboseHelp) {
     addTreeShakeIconsFlag();
     usesTargetOption();
-    usesFlavorOption();
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/commands/build_ios_framework.dart:45]: macOS doesn't support flavors, only add option in subclass `BuildIOSFrameworkCommand`

File: packages/flutter_tools/lib/src/commands/build_ios_framework.dart [hunk 4]
```diff
@@ -410,7 +474,6 @@ end
     Directory iPhoneBuildOutput,
     Directory simulatorBuildOutput,
     Directory modeDirectory,
-    Directory outputDirectory,
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/commands/build_ios_framework.dart:413]: This param was unused.

**Review by zanderso — APPROVED** [review]
LGTM

**Pull request merged**